### DISCUSSION TITLE
Focus of langium (newbie)

### DISCUSSION THREAD

**Discussion by anrl** [eclipse-langium/langium#1769] [Q&A] [answered]
Hi,

I have a general question on the focus on langium. It says it provides first-class support for language server protocol. Does it mean its main focus is developing IDE add-ons for language processing. I am looking to write a transpiler that would translate a DSL to JS or Python. I want to run it as a command-line tool. Is langium the wrong tool? I have used OhmJS with great success in the past. I am interested in exploring a new tool and it looks like langium is a good one! I just got the arithmetics example running, thanks to your help!!

**Comment by msujew** [comment, marked as answer]
Hey @anrl, thank you for the question!

> It says it provides first-class support for language server protocol. Does it mean its main focus is developing IDE add-ons for language processing. I am looking to write a transpiler that would translate a DSL to JS or Python.

Just because we want to provide the best IDE experience with Langium, doesn't mean Langium should not be used for other purposes as well. As said in our [project readme](https://github.com/eclipse-langium/langium/blob/main/README.md):

> The framework is an all-in-one solution for building programming languages, domain specific languages, code generators, interpreters and compilers. It serves as a spiritual successor to the [Eclipse Xtext framework](https://www.eclipse.org/Xtext/).

We have a [bunch](https://github.com/eclipse-langium/langium/tree/main/examples/arithmetics#interpreter-cli) [of](https://github.com/TypeFox/langium-lox/blob/main/langium/src/interpreter/cli.ts) [examples](https://github.com/eclipse-langium/langium/tree/main/packages/langium-cli) that show how to build CLIs with Langium. When using the [yeoman project generator](https://langium.org/docs/learn/workflow/scaffold/), you will be immediately prompted whether you want to have a CLI starter build included in your project.

**Reply by anrl** [reply]
Thanks a lot for the quick reply.

The yeoman project generator does not provide a CLI or VSCode extension option, which allows you to go with CLI without VSCode. It provides the CLI as a backup to the VSCode... that was my confusion. Thanks for clarifying it.

**Reply by msujew** [reply]
We have a PR in the works that attempts to split out all the parts of the generated project in https://github.com/eclipse-langium/langium/pull/1520. It's pretty massive, which is why it takes a while until we merge this.

**Reply by anrl** [reply]
I see. Thanks again!
